### DISCUSSION TITLE
Position of new tab

### DISCUSSION THREAD

**Discussion by hemanti0503** [piroor/treestyletab#3074] [Q&A]
I'm not sure if it's a matter of not finding the correct settings or if this is simply not in the scope of TST. What I'm trying to do is to open a new tab always on the top level. However, I don't want it to appear as the last tab, but before the "next" top-level tab.

Here is an example tree to illustrate what I mean:

- Tab 1
- Tab 2
  - Tab 2.1
  - Tab 2.2
- Tab 3
- Tab 4

Now some scenarios:
1. Tab 1 is active. A new tab should open right before tab 2.
2. Tab 2 is active. A new tab should open right above tab 3.
3. Tab 2.1 is active: A new tab should open right above tab 3.
4. Tab 2.2 is active: A new tab should open right above tab 3.

Setting behavior for new tab to "Direct neighbor on level of active tab" already fixes 1 and 2 for me. Now, is 3 and 4 also possible somehow?

**Comment by irvinm** [comment]
I don't think there are any options to accomplish exactly what you are trying to do.  I am pretty sure this would be possible with a helper addon to TST.  You could just monitor for new tabs, check the position and relationships of the current tree, then move the tab to the desired location.
